src/main/java/io/github/eliux/mega/DateBuilder.java

```diff
@@ -0,0 +1,55 @@
+package io.github.eliux.mega;
+
+import io.github.eliux.mega.error.MegaExpireDateRequiredException;
+import io.github.eliux.mega.error.MegaInvalidExpireDateException;
+
+import java.time.LocalDate;
+import java.time.Period;
+
+public class DateBuilder {
```

Comment:
Checking later the logic of this class I notice that the real purpose of this class is not actually to build a date. The real reason why you need this class is to build a date range, so instead of having a current date. You must have a `startDate `and  instead of expireDate you should have an `endDate`. So for a `DateRangeBuilder` or `LocalDateRangeBuilder` you have pretty much the same rules you need for the `ExportInfo`, but in addition it seems more reusable for any other context that needs a date range. Also instead of a builder you can create something as easy like a factory pattern which seems easier:
```
LocalDateRange.for(startDate, endDate);
// or
LocalDateRange.parse(startDateStr, endDateStr);
```
So this would make sense that instead of having an Optional of a builder you have an Optional of an specific data structure: `Optional<DateRange>`.